### PR TITLE
Restructured project.

### PR DESCRIPTION
**Pull request by MTrent1**
Moved interpreter and BinOps enum into a separate file from main function and put the new file inside a new folder.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -1,636 +1,10 @@
...
-            MetadataOperand => todo!(),
-        }
-    }
-}
+use llvm_ir::Module;
+use std::env;
 
 fn main() {
-    let mut bf = BFInterp::new();
+    let mut bf = llvm_ir_interpreter::BFInterp::new();
```
Comment (ptersilie) [inline src/main.rs]: We should also update `BFInterp`. And maybe we can shorten `llvm_ir_interpreter` too (maybe just `interp`)? Maybe the following structure would work:
```
main.rs
interp/
    mod.rs
```
Thoughts?

File: src/llvm_ir_interpreter.rs [hunk 2]
```diff
@@ -20,14 +20,14 @@ enum BinOps {
     Div,
     Rem,
 }
-pub(crate) struct BFInterp {
+pub(crate) struct LlvmIrInterpreter {
```
Comment (ptersilie) [inline src/llvm_ir_interpreter.rs]: LLVM and IR are typically capitalised so it should be `LLVMIRInterpreter`.

File: src/main.rs [hunk 3]
```diff
@@ -4,10 +4,10 @@ use llvm_ir::Module;
 use std::env;
 
 fn main() {
-    let mut bf = llvm_ir_interpreter::BFInterp::new();
+    let mut llvm_ir_interpreter = llvm_ir_interpreter::LlvmIrInterpreter::new();
```
Comment (ptersilie) [inline src/main.rs]: I would just shorten the variable name to `interp` or `lii`, or similar.

**Comment by ptersilie** [comment]
Two comments. There's also a typo in the first commit.

**Comment by MTrent1** [comment]
Ready for re-review.

**Comment by ptersilie** [comment]
Please squash. You can also update the commit message when you do.

**Comment by MTrent1** [comment]
Squashed.

**Comment by ptersilie** [comment]
Something gone wrong during your squashing. You've somehow included the commit from the previous PR in this one.

**Comment by MTrent1** [comment]
Whoops. Isn't the previous commit supposed to have already been there?

**Comment by MTrent1** [comment]
Fixed.

**Comment by ptersilie** [comment]
That looks better. I forgot to mention though that when moving a file, the move should be kept as a separate commit. Because now we have the move squashed together with other changes which are now almost impossible to find. It's too late for that now, but I thought I mention it for the future.

bors r+

**Pull request merged**